src/main/java/io/jenkins/plugins/forensics/miner/CommitStatistics.java

```diff
@@ -132,7 +132,7 @@ public String toString() {
 
     private static int getDistinctCount(final Collection<? extends CommitDiffItem> commits,
             final Function<CommitDiffItem, String> property) {
-        return (int) commits.stream().map(property).map(s -> String.toLowerCase(Locale.ENGLISH)).distinct().count();
+        return (int) commits.stream().map(property).map(s -> property.toString().toLowerCase()).distinct().count();
```

Comment:
This is wrong, and that the reason you needed to change the existing tests 🙈 

You are counting the number of different properties now! But you need to count the values!!!

```suggestion
        return (int) commits.stream().map(property).map(s -> s.toLowerCase(Locale.ENGLISH)).distinct().count();
```

And do not forget to add the import for Locale? Are you not coding in an IDE?

Reply:
> This is wrong, and that the reason you needed to change the existing tests 🙈
> 
> You are counting the number of different properties now! But you need to count the values!!!
> 
> And do not forget to add the import for Locale. Are you not coding in an IDE?

I am sorry for the writing the wrong code. I do not know how to write test cases in Java. I want to learn. The documentation does not match anything. Jenkins code gets complex at times.

If your schedule allows would you be my mentor? For a few weeks, say 3-4 weeks, or whatever fits your schedule. I promise I would not waste your time. 

I can research and learn on my own. I need help in understanding which part of code does what and in writing test cases.

Reply:
I can guide you for *this* specific issue and associated test. For additional things I have no time right now.

Reply:
> I can guide you for _this_ specific issue and associated test. For additional things I have no time right now.

Thank you for guiding me for this issue.

Reply:
I will start with the test case first.

Reply:
Hey @uhafner 

I wrote the test case. Reverted all the old changes and fixed the the code to resolve the issue.